File: config/src/main/java/de/eschoenawa/lanchat/config/persist/ConfigPersister.java
package de.eschoenawa.lanchat.config.persist;

import de.eschoenawa.lanchat.config.Config;

import java.util.Map;

public interface ConfigPersister {
    void save(Map<String, Config.Setting> settings);
    void load(ConfigPersisterLoadCallback callback);

    interface ConfigPersisterLoadCallback {
        void onSettingLoaded(String key, String value);
        void onPersistentSettingsNotYetCreated();
        void onLoadFailed(String reason);
        void onLoadDone();
    }
}

File: config/src/main/java/de/eschoenawa/lanchat/config/persist/ConfigPersister.kt
package de.eschoenawa.lanchat.config.persist

import de.eschoenawa.lanchat.config.Config

interface ConfigPersister {
    fun save(settings: Map<String, Config.Setting>)
    fun load(callback: ConfigPersisterLoadCallback)
    interface ConfigPersisterLoadCallback {
        fun onSettingLoaded(key: String, value: String)
        fun onPersistentSettingsNotYetCreated()
        fun onLoadFailed(reason: String)
        fun onLoadDone()
    }
}